src/Database/LSMTree/Internal/Index/Compact/Construction.hs

```diff
@@ -75,7 +76,7 @@ data MCompactIndex s = MCompactIndex {
     -- | Accumulates chunks of 'ciClashes'.
   , mciClashes              :: !(STRef s (NonEmpty (VU.MVector s Bit)))
     -- | Accumulates the 'ciTieBreaker'.
-  , mciTieBreaker           :: !(STRef s (Map SerialisedKey PageNo))
+  , mciTieBreaker           :: !(STRef s (Map (Unsliced SerialisedKey) PageNo))
```

Comment:
So my main question about the whole approach is why we need it in the first place rather than just using an existing type like `ShortByteString`, and I guess this here is the main argument in favour: it has a phantom type that shows what it's for, i.e. keys in this case.

Reply:
If we used `ShortByteString` we'd still have to write some functions like `fromRawBytes` and co that do the copying/asserting

Reply:
But yes the type parameter is helpful